mitodl/mitxonline
# add reference_number to Order model, backfill and test

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/799

#### What's this PR do?
Adding reference_number field to the Order model, also includes DB migration and backfill

#### How should this be manually tested?
The automated test should cover it. You could also
- Run the DB migration
- Enroll in a course in frontend
- Check the order edit page on django admin

<img width="774" alt="image" src="https://user-images.githubusercontent.com/3138890/183744256-72d284e8-a641-41c2-a9d5-f514b48ed2aa.png">
